src/settings-tab.ts

```diff
@@ -112,22 +112,36 @@ export class MindMapSettingsTab extends PluginSettingTab {
           })
       );
 
+    function decide_display_colors(approach: MindMapSettings['coloring']) {
```

Comment:
- change the function name to camelCase
- move it to the top (make it another method)
- Modify it so this function receives an array of Setting (for dynamic colors), a single one Settings for the default one, and the approach.